### PR TITLE
Protocol-spec multi-signature registration - Closes #4185

### LINKED ISSUE #4185: Protocol-spec multisignature registration
### Description

- Basic Multisignature acceptance
- Multisignature registration and funding of account for it on same block
- Multisignature registration and transfer from multisignature
- Combination of multi signature and transfer transactions in a same block

### Acceptance Criteria

JSON generator for: 

- [x] Basic Multisignature acceptance
- [x] Multisignature registration and funding of account for it on same block
- [x] Multisignature registration and transfer from multisignature
- [x] Multisignature and transfer transactions in a same block

### Additional Information

Split from https://github.com/LiskHQ/lisk-sdk/issues/4087

### PR DESCRIPTION
**Pull request by pablitovicente**
### What was the problem?

Protocol-specs generators for multi-signatures did not exist 

### How did I solve it?

- by adding helper methods for generating multi-signature transactions
- by adding the generator

### Review checklist

- [x] The PR resolves #4185
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

### CODE REVIEW

File: protocol-specs/generators/block_processing_multisignatures/index.js [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+		.transfer('100')
+		.from('16313739661670634666L')
+		.to('10881167371402274308L')
+		.forge();
+	// Fund three accounts
+	chainStateBuilder
+		.transfer('30')
+		.from('10881167371402274308L')
+		.to('8465920867403822059L')
+		.transfer('30')
```
Comment (shuse2) [inline protocol-specs/generators/block_processing_multisignatures/index.js]: do we need 3 transfer here? I think one is sufficient to make it fail

File: protocol-specs/generators/block_processing_multisignatures/index.js [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+
+	// Transfer funds from genesis account to one of the delegates
+	chainStateBuilder
+		.transfer('100')
+		.from('16313739661670634666L')
+		.to('10881167371402274308L')
+		.forge();
+	// Fund three accounts
+	chainStateBuilder
+		.transfer('30')
```
Comment (shuse2) [inline protocol-specs/generators/block_processing_multisignatures/index.js]: why do we need to fund these 3 accounts?

File: protocol-specs/generators/block_processing_multisignatures/index.js [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+	handler:
+		'invalid_block_processing_multisignature_registration_and_funding_for_members_same_block',
+	testCases: generateTestCasesInvalidBlockMultisignatureRegistrationAndFundingInSameBlock(),
+});
+
+const invalidBlockWithTransferFromMultisigAccountOnSameBlockAsRegistration = () => ({
+	title: 'Invalid block processing',
+	summary:
+		'An invalid block with a multisignature registration and transfer from that account on same block',
+	config: 'mainnet',
```
Comment (shuse2) [inline protocol-specs/generators/block_processing_multisignatures/index.js]: I just realized but why we use `mainnet` config?

File: protocol-specs/generators/block_processing_multisignatures/index.js (hunk 1 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_multisignatures/index.js]: That's correct it's an error I will leave just one.

File: protocol-specs/generators/block_processing_multisignatures/index.js (hunk 2 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_multisignatures/index.js]: Same as above.

File: protocol-specs/generators/block_processing_multisignatures/index.js (hunk 3 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_multisignatures/index.js]: I was just using it from the previous implementation. I don't know if that value is currently being used at all.

File: protocol-specs/generators/block_processing_multisignatures/index.js (hunk 1 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_multisignatures/index.js]: Fixed.

File: protocol-specs/generators/block_processing_multisignatures/index.js (hunk 2 above)
Comment (pablitovicente) [reply protocol-specs/generators/block_processing_multisignatures/index.js]: Fixed

**Pull request merged**